### PR TITLE
[requested-tool] : PKGBUILD for crlfuzz

### PR DESCRIPTION
**Pull request by D4imyo**

### CODE REVIEW

File: packages/crlfuzz/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# This file is part of BlackArch Linux ( https://www.blackarch.org/ ).
+# See COPYING for license details.
+
+pkgname=crlfuzz
+pkgver=v1.4.1.r4.gc960f98
+pkgrel=1
+pkgdesc='A fast tool to scan CRLF vulnerability written in Go'
```
Comment (Edu4rdSHL) [inline packages/crlfuzz/PKGBUILD:7]: Add final dot.
```suggestion
pkgdesc='A fast tool to scan CRLF vulnerability written in Go.'
```

File: packages/crlfuzz/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+url='https://github.com/dwisiswant0/crlfuzz'
+groups=('blackarch' 'blackarch-webapp')
+arch=('any')
+source=('git+https://github.com/dwisiswant0/crlfuzz.git')
```
Comment (Edu4rdSHL) [inline packages/crlfuzz/PKGBUILD:11]: Use variables whatever you can.
```suggestion
source=("git+https://github.com/dwisiswant0/$pkgname.git")
```

File: packages/crlfuzz/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# This file is part of BlackArch Linux ( https://www.blackarch.org/ ).
+# See COPYING for license details.
+
+pkgname=crlfuzz
+pkgver=v1.4.1.r4.gc960f98
+pkgrel=1
+pkgdesc='A fast tool to scan CRLF vulnerability written in Go'
+url='https://github.com/dwisiswant0/crlfuzz'
+groups=('blackarch' 'blackarch-webapp')
+arch=('any')
```
Comment (Edu4rdSHL) [inline packages/crlfuzz/PKGBUILD:10]: Golang packages produce binaries, add specific architectures.
```suggestion
arch=('x86_64' 'aarch64')
```

File: packages/crlfuzz/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+source=('git+https://github.com/dwisiswant0/crlfuzz.git')
+makedepends=('go')
+license=('MIT')
+sha512sums=('SKIP')
+
+pkgver(){
+  cd $pkgname
+  git describe --long --tags | sed 's/\([^-]*-g\)/r\1/;s/-/./g'
```
Comment (Edu4rdSHL) [inline packages/crlfuzz/PKGBUILD:18]: Always leave an empty line between sections like `cd` and other commands after it.
```suggestion
  cd $pkgname
  
  git describe --long --tags | sed 's/\([^-]*-g\)/r\1/;s/-/./g'
```

File: packages/crlfuzz/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+build(){
+  cd $pkgname/cmd/$pkgname
+  go build .
+}
```
Comment (Edu4rdSHL) [inline packages/crlfuzz/PKGBUILD:24]: Follow our [build guidelines](https://github.com/BlackArch/blackarch-pkgbuilds/blob/9e4dffa5e8f5f74374a2a31daf8411a361a51376/PKGBUILD-go#L27-L43).

File: packages/crlfuzz/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  install -dm 755 "$pkgdir/usr/share/$pkgname/"
+  install -Dm 755 "cmd/$pkgname/$pkgname" "$pkgdir/usr/bin/$pkgname"
+
+  install -Dm 644 LICENSE "$pkgdir/usr/share/license/$pkgname/LICENSE"
+  install -Dm 644 -t "$pkgdir/usr/share/doc/$pkgname/" *.md
+
+  rm LICENSE *.md
+
+  cp -a * "$pkgdir/usr/share/$pkgname/"
+}
```
Comment (Edu4rdSHL) [inline packages/crlfuzz/PKGBUILD:38]: Add EOF.

**Review by Edu4rdSHL — CHANGES_REQUESTED** [review]
Hello, thank you so much for it. Please fix it according to comments, also **always** follow our [PKGBUILD styles](https://github.com/BlackArch/blackarch-pkgbuilds). For Golang, [see it](https://github.com/BlackArch/blackarch-pkgbuilds/blob/master/PKGBUILD-go).

File: packages/crlfuzz/PKGBUILD [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+pkgname=crlfuzz
+pkgver=v1.4.1.r4.gc960f98
+pkgrel=1
+pkgdesc='A fast tool to scan CRLF vulnerability written in Go'
+url='https://github.com/dwisiswant0/crlfuzz'
+groups=('blackarch' 'blackarch-webapp')
+arch=('any')
+source=('git+https://github.com/dwisiswant0/crlfuzz.git')
+makedepends=('go')
```
Comment (Edu4rdSHL) [inline packages/crlfuzz/PKGBUILD:12]: Add `git` to makedepends. If you build in a clean chroot environment you have noticed it.

**Review by Edu4rdSHL — CHANGES_REQUESTED** [review]

File: packages/crlfuzz/PKGBUILD [hunk 8]
```diff
@@ -4,30 +4,39 @@
...
-pkgdesc='A fast tool to scan CRLF vulnerability written in Go'
+pkgdesc='A fast tool to scan CRLF vulnerability written in Go.'
 url='https://github.com/dwisiswant0/crlfuzz'
 groups=('blackarch' 'blackarch-webapp')
-arch=('any')
-source=('git+https://github.com/dwisiswant0/crlfuzz.git')
-makedepends=('go')
+arch=('x86_64' 'arch64')
+source=("git+https://github.com/dwisiswant0/$pkgname.git")
+makedepends=('go' 'git')
```
Comment (Edu4rdSHL) [inline packages/crlfuzz/PKGBUILD:12]: Put makedepends before source.

File: packages/crlfuzz/PKGBUILD (hunk 6 above)
Comment (Edu4rdSHL) [reply packages/crlfuzz/PKGBUILD:47]: It's still missing.

**Review by Edu4rdSHL — CHANGES_REQUESTED** [review]

**Review by Edu4rdSHL — APPROVED** [review]
Hey @D4imyo, thank you so much for it, really appreciated, splendid work. I will merge it once we finish an internal updating process!

**Comment by noptrix** [comment]
hey guys, i didn't see this before and i commite dcrlfuzz already. closing issue

**Comment by noraj** [comment]
added in e33adf69c4cf569462a37061bf983987cb6ba8fa

**Pull request closed without merging**